tests/bench/bench_single_conn_test.go

```diff
@@ -0,0 +1,156 @@
+
+	cluster.Dialer = recorder.NewRecordDialer(dir)
+
+	fallback := gocql.RoundRobinHostPolicy()
+	cluster.PoolConfig.HostSelectionPolicy = gocql.TokenAwareHostPolicy(fallback)
+
+	executor, err := gocql.NewSingleHostQueryExecutor(cluster)
+	if err != nil {
+		panic(err)
+	}
+	defer executor.Close()
+
+	for i := 0; i < size; i++ {
+		iter := executor.Iter(`SELECT v FROM single_conn_bench.table1 WHERE pk = ?;`, i)
+		var name string
+		for iter.Scan(&name) {
+			if name[:4] != "Name" {
+				panic("Wrong value")
+			}
+		}
+		if err := iter.Close(); err != nil {
+			panic(err)
+		}
+	}
+}
+
+func RecordInsertTraffic(size int, dir string) {
+	cluster := gocql.NewCluster("192.168.100.11")
+	cluster.Consistency = gocql.Quorum
+
+	cluster.Dialer = recorder.NewRecordDialer(dir)
+
+	fallback := gocql.RoundRobinHostPolicy()
+	cluster.PoolConfig.HostSelectionPolicy = gocql.TokenAwareHostPolicy(fallback)
+
+	executor, err := gocql.NewSingleHostQueryExecutor(cluster)
+	if err != nil {
+		panic(err)
+	}
+	defer executor.Close()
+
+	for i := 0; i < size; i++ {
+		err = executor.Exec(`INSERT INTO single_conn_bench.table1 (pk, ck, v) VALUES (?, ?, ?);`, i, i%5, fmt.Sprintf("Name_%d", i))
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+func BenchmarkSingleConnectionSelect(b *testing.B) {
+	cluster := gocql.NewCluster("192.168.100.11")
+	cluster.Consistency = gocql.Quorum
+
+	cluster.Dialer = replayer.NewReplayDialer("rec_select")
+
+	fallback := gocql.RoundRobinHostPolicy()
+	cluster.PoolConfig.HostSelectionPolicy = gocql.TokenAwareHostPolicy(fallback)
+
+	executor, err := gocql.NewSingleHostQueryExecutor(cluster)
+	if err != nil {
+		panic(err)
+	}
+	defer executor.Close()
+
+	b.Run("Select", func(b *testing.B) {
+		for i := 0; i < 10; i++ {
+			b.Run("Case"+strconv.Itoa(i), func(b *testing.B) {
+				for j := 0; j < b.N; j++ {
+					_ = executor.Iter(`SELECT v FROM single_conn_bench.table1 WHERE pk = ?;`, i)
+				}
+			})
+		}
+	})
+}
+
+func BenchmarkSingleConnectionInsert(b *testing.B) {
+	cluster := gocql.NewCluster("192.168.100.11")
+	cluster.Consistency = gocql.Quorum
+
+	cluster.Dialer = replayer.NewReplayDialer("rec_insert")
+
+	fallback := gocql.RoundRobinHostPolicy()
+	cluster.PoolConfig.HostSelectionPolicy = gocql.TokenAwareHostPolicy(fallback)
+
+	executor, err := gocql.NewSingleHostQueryExecutor(cluster)
+	if err != nil {
+		panic(err)
+	}
+	defer executor.Close()
+
+	b.Run("Insert", func(b *testing.B) {
+		for i := 0; i < 10; i++ {
+			b.Run("Case"+strconv.Itoa(i), func(b *testing.B) {
+				for j := 0; j < b.N; j++ {
+					err = executor.Exec(`INSERT INTO single_conn_bench.table1 (pk, ck, v) VALUES (?, ?, ?);`, i, i%5, fmt.Sprintf("Name_%d", i))
+					if err != nil {
+						panic(err)
+					}
+				}
+			})
+		}
+	})
+}
+
+func init() {
+	InitializeCluster()
+	RecordInsertTraffic(10, "/home/sylwiaszunejko/gocql/tests/bench/rec_insert")
+	RecordSelectTraffic(10, "/home/sylwiaszunejko/gocql/tests/bench/rec_select")
+}
```

Comment:
Please check if it is possible to make it run only if you do `go test --update-golden`.

Reply:
I tried to add a flag like in `recreate_test.go`, but it complains that the flag is not defined, pls remind me how to provide that flag in `recreate_test.go` to make it work? I tried `/integration.sh integration scylla --update-golden` but it does not work

Reply:
Try this:
```
func TestMain(m *testing.M) {
	update := flag.Bool("update-golden", false, "Update golden files")
	flag.Parse()
	if *update {
		InitializeCluster()
		RecordInsertTraffic(10, "rec_insert")
		RecordSelectTraffic(10, "rec_select")
	}
	os.Exit(m.Run())
}

```

Reply:
it works with this version